apps/arweave/src/ar_chunk_storage.erl

```diff
@@ -871,6 +873,29 @@ is_offset_valid(Byte, BucketStart, ChunkOffset) ->
 	Delta = Byte - (BucketStart + ChunkOffset rem ?DATA_CHUNK_SIZE),
 	Delta >= 0 andalso Delta < ?DATA_CHUNK_SIZE.
 
+filter_by_sync_record(Chunks, _Byte, _Start, _ChunkFileStart, _StoreID, 1) ->
+	Chunks;
```

Comment:
Is this to handle the special case where we query a single chunk (e.g. from a code path other than the mining code path)? In that case we *won't* filter?

Reply:
In this case we have already implicitly checked the chunk belongs to the sync record - https://github.com/ArweaveTeam/arweave/blob/master/apps/arweave/src/ar_chunk_storage.erl#L168